### PR TITLE
Decide inline on calling to_table_display

### PR DESCRIPTION
**Pull request by benny-burkhart**
Remove the need for TableDisplay to be included in Enumerable and its descendants.
Remove the requirement to call `send(:include, TableDisplay)`.

### CODE REVIEW

**Comment by willbryant** [comment]
This seems like a fair option, but it would be a backwards incompatible change (your company had some emails that used to_table_display, for example), so I'd like to understand the options.  Have asked for some clarification on the issue.

Also the README would need to be revised if we took this approach.

File: lib/table_display.rb [hunk 1]
```diff
@@ -105,15 +105,6 @@ def to_table_display(*args)
 
 module Kernel
   def pt(target, *options)
-    puts target.respond_to?(:to_table_display) ? target.to_table_display(*options) : target
+    puts target.class.ancestors.include?(Enumerable) ? TableDisplay.to_table_display(target, *options) : target
```
Comment (willbryant) [inline lib/table_display.rb]: I think this would be normally implemented using `target.is_a?(Enumerable)` or perhaps just `target.respond_to?(:each)`.

Comment (benny-burkhart) [reply lib/table_display.rb]: Thanks, good point -- though `to_table_display` calls both `each` and `collect` on the target, so I've refactored the suitability test into its own method.

**Comment by benny-burkhart** [comment]
Thank you for your time.

> it would be a backwards incompatible change
Yes, this would warrant a new major version number I assume.

> (your company had some emails that used to_table_display, for example)
These are gone. Here this is in use to display results of failed tests, and ad hoc on the console.

> Also the README would need to be revised if we took this approach.
The rebase includes updated to the README and to the test.

Please get back as and when you find time.

**Comment by benny-burkhart** [comment]
> > (your company had some emails that used to_table_display, for example)
> 
> These are gone. Here this is in use to display results of failed tests, and ad hoc on the console.

In fact these are still present. They are easy to refactor though.

**Comment by benny-burkhart** [comment]
Thanks for your efforts!

**Pull request closed without merging**